pkg/shell/base_index.js

```diff
@@ -143,6 +144,42 @@ var phantom_checkpoint = phantom_checkpoint || function () { };
                 }
             }
 
+            function domChangeHandler(mutation) {
```

Comment:
Again, we're very near the Router code here. We could just use that to track things. Why do we need to use a mutation observer?